crm_compassion/model/demand_planning.py

```diff
@@ -33,43 +34,84 @@ class WeeklyDemand(models.Model):
         return res
 
     @api.multi
     def write(self, vals):
-        vals = self.update_vals(vals)
+        self.update_vals(vals)
         res = super(WeeklyDemand, self).write(vals)
         return res
 
     ##########################################################################
     #                             PUBLIC METHODS                             #
     ##########################################################################
     def update_vals(self, vals):
-        starting_date = datetime.strptime(vals.get('week_start_date'),
-                                          "%Y-%m-%d")
-        if self.check_is_date_in_next_8_weeks(starting_date):
-            vals['period_locked'] = True
-        return vals
+        starting_date = vals.get('week_start_date')
+        if starting_date:
+            corrected_date = datetime.strptime(starting_date, "%Y-%m-%d")
+            if self.check_is_date_in_next_8_weeks(corrected_date):
+                vals['period_locked'] = True
+            else:
+                vals['period_locked'] = False
+        # return vals
 
     def check_is_date_in_next_8_weeks(self, date):
         return date <= (datetime.today() + timedelta(weeks=8))
 
 
 class DemandPlanning(models.Model):
-
     _name = 'demand.planning'
     _description = 'Demand Planning'
 
     name = fields.Char(required=True)
     date = fields.Date()
     sent_date = fields.Datetime()
     state = fields.Selection([
-        ('new', 'New'),
+        ('draft', 'Draft'),
         ('sent', 'Sent'),
         ('error', 'Error'),
     ])
     weekly_demand_ids = fields.Many2many(
         'demand.weekly.demand', string='Weekly Demands'
     )
+
+    ##########################################################################
+    #                             ORM METHODS                                #
+    ##########################################################################
+    @api.model
+    def create(self, vals):
+        res = super(DemandPlanning, self).create(vals)
+        res.state = 'draft'
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(DemandPlanning, self).write(vals)
+        if not isinstance(res, bool):
+            res.state = 'draft'
+        return res
+
+    ##########################################################################
+    #                             PUBLIC METHODS                             #
+    ##########################################################################
+    @api.multi
+    def send_planning(self):
+        message_obj = self.env['gmc.message.pool']
+
+        action_id = self.env.ref(
+            'crm_compassion.create_demand_planning').id
+        object_id = self.id
+        message_vals = {
+            'action_id': action_id,
+            'object_id': object_id
+        }
+        pool = message_obj.with_context(async_mode=False,
+                                        creating=True).create(message_vals)
```

Comment:
`creating=True` is not relevant anymore since I added the button that triggers the message sending. I forgot the remove it.